akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -62,6 +62,7 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
   private var writeInProgress = false
   private var sequenceNr: Long = 0L
   private var _lastSequenceNr: Long = 0L
+  private var withinHandler = false
```

Comment:
This requires some thinking regarding binary compatibility. Since this is a trait we can normally not change much of it. My thinking is that it might be alright here. If a library is compiled with an earlier Akka version this is not included in those class files, but the methods using this field are also in the same trait and would also not be the library (well, old versions of the methods would).

Reply:
I'll test it out

Reply:
I'll try to reproduce the exact case since users don't extend Eventsourced but PersistentActor...

But in general the problem is that `private var` still is a public method on the interface generated:

```
$ cat Example.scala
trait Example {
  private var num = 21
  def lol() = ""

}
ktoso @ 三日月/tmp
(reverse-i-search)`ja': java -cp $SCALA_LIB_2_12:. User
ktoso @ 三日月/tmp
$ javap -cp . -p Example
Compiled from "Example.scala"
public interface Example {
  public abstract int Example$$num(); // OH NOES!!!
  public abstract void Example$$num_$eq(int);// OH NOES!!!

  public static java.lang.String lol$(Example);
  public java.lang.String lol();
  public static void $init$(Example);
}
```

So when someone extended it, back then, and now would use it with the new Example on classpath they'd get:

```
$ java -cp $SCALA_LIB_2_12:. User
Exception in thread "main" java.lang.AbstractMethodError: Method User.Example$$num_$eq(I)V is abstract
```

I'll see if our layer of PersistentActor is helping here or not though

Reply:
I'm sad to report that this will blow up in this situation:

```
$ cat Example.scala
trait Example {
  private var num = 21
  def lol() = "hello  "

}
ktoso @ 三日月/tmp
$ cat PA.scala
trait PA extends Example {}
ktoso @ 三日月/tmp
$ echo "PA and Example are compiled together"
PA and Example are compiled together
ktoso @ 三日月/tmp
$
ktoso @ 三日月/tmp
$ echo User.scala
User.scala
ktoso @ 三日月/tmp
$ java -cp $SCALA_LIB_2_12:. User
Exception in thread "main" java.lang.AbstractMethodError: Method User.Example$$num_$eq(I)V is abstract
	at User.Example$$num_$eq(User.scala)
	at Example.$init$(Example.scala:2)
```

So creating an PersistentActor by users (User) would become broken if we do this.